appsec/src/helper/service.hpp

```diff
@@ -75,6 +75,11 @@ class service {
         return schema_sampler_;
     }
 
+    void set_sampler_rate(double rate)
+    {
+        schema_sampler_ = std::make_shared<sampler>(rate);
```

Comment:
Because now `schema_sampler_` can be written to from different threads, you need to have a [`std::atomic<std::shared_ptr<sampler>>`](https://en.cppreference.com/w/cpp/memory/shared_ptr/atomic2) (for C++20).

Reply:
Done